R/themes.R

```diff
@@ -35,7 +36,37 @@ addTheme <- function(themePath,
                      force = FALSE,
                      globally = FALSE)
 {
-  callFun("addTheme", themePath, apply, force, globally)
+  path <- themePath
+
+  # Ensure path looks like something we can use
+  ext <- tolower(tools::file_ext(path))
+  if (!identical(ext, "rstheme") && !identical(ext, "tmtheme")) {
+    stop("Invalid path ", path, ". ",
+         "Please supply a path or URL to an .rstheme or .tmtheme file to add.")
+  }
+
+  # If the path appears to be a URL, download it.
+  if (grepl("^https?:", themePath)) {
+    # Give the downloaded filename the same name and extension as the original.
+    parts <- strsplit(themePath, "/", fixed = TRUE)[[1]]
```

Comment:
Nit: I think `basename()` would still work fine for URLs.

Reply:
Good call.